R/validate_factory.R

```diff
@@ -1,143 +1,134 @@
-
 #' Inspect and validate the content of a factory
 #'
-#' This function can be used to inspect the content of a factory and make sure
-#' it looks fine. This includes various sanity checks listed in details. The
-#' function returns a list of error and warning messages.
+#' `validate_factory()` can be used to inspect the content of a factory and make
+#' everything looks fine. This includes various sanity checks listed in details
+#' that will error if a problem is found.
+#' 
+#' @inheritParams compile_reports
+#' @param allow_duplicates A logical indicating duplicate filenames are
+#'   allowed across folders.
 #'
 #' @details
-#' Checks ran on the factory include
-#' (the result of a failure is indicated in brackets):
-#'
-#' \itemize{
-#'
-#'  \item the directory exists [error]
-#'
-#'  \item all mandatory files exist: .here, .gitignore [error]
-#'
-#'  \item all mandatory folders exist: report_sources/ [error]
-#'
-#'  \item all .Rmd reports have unique names once outside their folders (to
-#'  avoid conflicts in outputs) [warning]
-#'
-#' }
-#'
+#' Checks run on the factory include:
+#'   * the factory directory exists;
+#'   * the factory_config file exist;
+#'   * all mandatory folders exist - by default these are 'report_sources/'
+#'     and 'outputs/';
+#'   * (optionally) all .Rmd reports have unique names once outside their
+#'     folders.
```

Comment:
Putting a pin in this for discussing, but also to chek. Assuming we have:
report_sources/foo.Rmd
report_sources/bar/foo.Rmd
Then, is the the case that:
`compile_reports("foo")`
will compile both but for any duplicated output (i.e. output with the same name), only the second version will be stored in `outputs/foo/[timestamp]`?
Or will it generate a mirrored output folder structure? If the latter is true, then we may no longer want to check that reports have unique names outside their folders.

Reply:
The latter is correct.  I've removed the duplicates check in https://github.com/reconhub/reportfactory/pull/68/commits/c6bba8180dba8e6a811501d54784583c36f5bc54